src/main/java/Duke.java

```diff
@@ -1,10 +1,342 @@
+            if (taskNumber > 0 && taskNumber <= listSize) {
+                //If valid task number is given.
+                list[taskNumber - 1].unmarkDone();
+                lineSeparator();
+                System.out.println("OK, I've marked this task as not done yet:");
+                System.out.println("  " + list[taskNumber - 1].toString());
+                lineSeparator();
+            } else {
+                //If task number given exceeds total tasks in the list.
+                throw new DukeException();
+            }
+        } else {
+            //If user types non-integer inputs after "unmark".
+            throw new DukeException();
+        }
+    }
+
+    /**
+     * Prints a string to inform the user that the program is terminating.
+     *
+     * @param command The string array containing all individual strings separated by a space (" ") character in
+     *                the user inputted string.
+     * @throws DukeException The custom defined exception relating to all errors specific to the Duke program.
+     */
+    private static void goodbyeMessage(String[] command) throws DukeException {
+        if (command.length == 1) {
+            lineSeparator();
+            System.out.println("Bye. Hope to see you again soon!");
+            lineSeparator();
+        } else {
+            //If 'bye' is not the only thing typed.
+            throw new DukeException();
+        }
+    }
+
+    /**
+     * Prints a series of error message strings depending on the wrong or partially wrong input by the user.
+     *
+     * @param line The single line of string inputted by the user.
+     */
+    private static void handleException(String line) {
+        String[] command = line.split(" ");
+        switch (command[0]) {
+        case "list":
+            lineSeparator();
+            System.out.println("OOPS!!! A 'list' command cannot contain a second word!");
+            lineSeparator();
+            break;
+        case "todo":
+            lineSeparator();
+            System.out.println("OOPS!!! A description of a 'todo' cannot be empty. Please follow this format below:");
+            System.out.println("    todo <INSERT TASK NAME>");
+            lineSeparator();
+            break;
+        case "event":
+            lineSeparator();
+            System.out.println("OOPS!!! Did you declare your 'event' properly? Please follow this format below:");
+            System.out.println("    event <INSERT TASK NAME> /from <INSERT START DATE> /to <INSERT END DATE>");
+            lineSeparator();
+            break;
+        case "deadline":
+            lineSeparator();
+            System.out.println("OOPS!!! Did you declare your 'deadline' properly? Please follow this format below:");
+            System.out.println("    deadline <INSERT TASK NAME> /by <INSERT END DATE>");
+            lineSeparator();
+            break;
+        case "mark":
+        case "unmark":
+            printInvalidInputError();
```

Comment:
The level of abstraction does not seem consistent in this function where in previous cases what is to be outputted is explicitly declared whereas for this another function is called. Try to standardize to a single level of abstraction where possible.